### PR TITLE
update history for Mesa 2.1 release

### PR DESCRIPTION
**Pull request by tpike3**
Updates history for 2.0 to account visualization becoming part of core via experimental

### CODE REVIEW

**Comment by rht** [comment]
The version bump has already been done. This needs to be 2.0.1, and with a note to tell people that they should ready 2.0.1 release note.

**Comment by rht** [comment]
Actually, it needs to be 2.1, given that `mesa.experimental` is a newly added namespace.

**Comment by jackiekazil** [comment]
Need to update the date as well -- `2.0.0 (2023-04-15) Wellton`

@rht is there a reason we wouldn't just bump the tag for 2.0? We never successfully released 2.0 on pypi. Not sure i understand the logic of why to make it a 2.1.

**Comment by rht** [comment]
From the Git revision standpoint, the version 2.0 has already happened. Someone browsing the revisions will be confused by the fact that subsequent commits are part of 2.0 instead of the next release.

**Comment by Corvince** [comment]
It is also possible to edit releases, see 

https://stackoverflow.com/questions/49897509/how-can-i-add-more-recent-commits-into-an-older-github-release

The answer mentions that it is very inconvenient for people that already forked/cloned the repo with the release(tag). Don't know if that applies to anyone at all. 2.1 would be safer, but I think we should also be fine to edit the release.

**Comment by tpike3** [comment]
Update history

* fix month (can't believe I missed that)
* create 2.1 (youngtown) milestone
* link commits to 2.1 milestone
* remove 2,1 commits from 2.0 history
* create 2.1 history

**release date is set for tomorrow to account for any edits and schedules**

*interesting to note the pull request title didn't change even though I squashed commits and changes the commit message; changed it manually

File: HISTORY.rst
Comment (rht) [inline HISTORY.rst:9]: Needs to say that since this is released right after, user needs to read 2.0.0 release note.

**Review by rht — APPROVED** [review]

**Comment by rht** [comment]
@tpike3 you forgot to bump the version in `mesa/__init__.py` to 2.1.0.

Waiting for approval from @jackiekazil and @Corvince. Also need to test-submit to https://test.pypi.org/ just to be sure.

**Review by Corvince — APPROVED** [review]

**Comment by tpike3** [comment]
> @tpike3 you forgot to bump the version in `mesa/__init__.py` to 2.1.0.
> 
> Waiting for approval from @jackiekazil and @Corvince. Also need to test-submit to https://test.pypi.org/ just to be sure.

Awww son of .... ... done

**Comment by rht** [comment]
At least the error message seems different. Can't know for sure without going all the way through:
```
Uploading Mesa-2.1-py3-none-any.whl
100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB • 00:09 • 228.1 kB/s
WARNING  Error during upload. Retry with the --verbose option for more details.                
ERROR    HTTPError: 403 Forbidden from https://test.pypi.org/legacy/                           
         The user 'rht' isn't allowed to upload to project 'Mesa'. See                         
         https://test.pypi.org/help/#project-name for more information.   
```

**Comment by Corvince** [comment]
I think it just means you don't have the necessary permissions on test.pypi.org  - they have to be set separately from pypi.org. I think @jackiekazil has to fix the permissions and/or upload

**Pull request merged**
